src/OpenLoco/src/GameCommands/VehicleReverse.cpp

```diff
@@ -0,0 +1,58 @@
+#include "Economy/Expenditures.h"
+#include "Entities/EntityManager.h"
+#include "GameCommands.h"
+#include "Types.hpp"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004ADAA8
+    static uint32_t vehicleReverse(EntityId headId, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        auto* head = EntityManager::get<Vehicles::VehicleHead>(headId);
+        if (head == nullptr)
+        {
+            return FAILURE;
+        }
+
+        if (!sub_431E6A(head->owner))
+        {
+            return FAILURE;
+        }
+
+        auto* head2 = EntityManager::get<Vehicles::VehicleHead>(head->head);
+        if (!head2->canBeModified())
+        {
+            return FAILURE;
+        }
+
+        if (static_cast<uint16_t>(head2->tileX) == 0xFFFF)
+        {
+            setErrorText(StringIds::empty);
+            return FAILURE;
+        }
+
+        if (!(flags & Flags::apply))
```

Comment:
the old code would have still called the setPosition on query.

Reply:
I deliberately omitted that, actually. If we're just querying, shouldn't that leave the state untouched?

Reply:
We don't do this consistently, though, so we should probably document that somewhere...